include/xtypes/idl/generator.hpp

```diff
@@ -104,8 +106,120 @@ inline std::string type_name(const DynamicType& type)
     }
     else
     {
-        return "[[unsupported]]";
+        //return "[[unsupported]]";
+        return type.name();
+    }
+}
+
+inline std::string structure(const StructType& type, size_t tabs = 0)
+{
+    std::stringstream ss;
+    ss << std::string(tabs * 4, ' ') << "struct " << type.name() << std::endl;
+    ss << std::string(tabs * 4, ' ') << "{" << std::endl;
+
+    for(const Member& member: type.members())
+    {
+        ss << std::string((tabs + 1) * 4, ' ');
+        if(member.type().kind() == TypeKind::ARRAY_TYPE)
+        {
+            ss << generator::array_member(member); //Spetial member syntax
+        }
+        else
+        {
+            ss << generator::type_name(member.type()) << " " << member.name() << ";";
+        }
+        ss << std::endl;
+    }
+    ss << std::string(tabs * 4, ' ') << "};" << std::endl;
+    return ss.str();
+}
+
+inline std::string enumeration32(const EnumerationType<uint32_t>& enumeration, size_t tabs = 0)
+{
+    std::stringstream ss;
+    // We must add them in order
+    using map_pair = std::pair<std::string, uint32_t>;
+    std::map<std::string, uint32_t> enumerators = enumeration.enumerators();
+    ss << std::string(tabs * 4, ' ') << "enum " << enumeration.name() << std::endl;
+    ss << std::string(tabs * 4, ' ') << "{" << std::endl;
+    // Copy to a vector
+    std::vector<map_pair> ids(enumerators.begin(), enumerators.end());
+    // Sort the vector
+    std::sort(ids.begin(), ids.end(),
+              [](const map_pair& a, const map_pair& b)
```

Comment:
Why 10 spaces of indentation?

Reply:
Because nobody tells to *Qt Creator* how many spaces to indent after splitting a list of arguments.
```suggestion
        [](const map_pair& a, const map_pair& b)
```

Reply:
:laughing: